### ISSUE TITLE
[Discussion]How to simplify kubeflow architecture

### ISSUE THREAD

**Issue by cndaimin** [kubeflow/kubeflow#2829]
Kubeflow is a bit confused me with a lot projects currently, is it in the plan to simplify the architecture? I am not quite sure whether my immature thoughts make sense below.

* Kubeflow server
    * scheduler: tensorflow, pytorch, mpi, notebook
    * rest api
* Kubeflow client(using rest api)
    * of different languages: python, java, go
    * web-front
    * command line tool based on go client

**Comment by jlewi** [comment]
This question comes up a lot. I'd suggest looking at this deck.
http://bit.ly/kf_landscape_2019

TL;DR We are trending in the direction of a loosely coupled microservice-architecture. I think a monolithic architecture presents scalability challenges

**Comment by cndaimin** [comment]
I got it, thanks @jlewi

**Comment by nicklausZJF** [comment]
> This question comes up a lot. I'd suggest looking at this deck.
> http://bit.ly/kf_landscape_2019
> 
> TL;DR We are trending in the direction of a loosely coupled microservice-architecture. I think a monolithic architecture presents scalability challenges

Thank you ! It is what I need, but I want more details about kubeflow architecture, do you have any other illustration ?
